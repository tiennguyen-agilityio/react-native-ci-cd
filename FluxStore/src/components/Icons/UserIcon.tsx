import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';

export const UserIcon = memo(
  ({
    width = 20,
    height = 24,
    color,
    disabled = false,
    isActive = false,
    style,
    onPress,
  }: IconProps) => {
    const {
      theme: {text, icon},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="user-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 20 24" fill="none">
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M19.783 18.892c0 3.846-5.462 4.29-9.57 4.29H9.92c-2.617-.006-9.277-.166-9.277-4.313 0-3.767 5.242-4.27 9.318-4.29h.547c2.616.006 9.275.166 9.275 4.313zm-9.57-2.563c-5.148 0-7.758.854-7.758 2.54 0 1.701 2.61 2.563 7.759 2.563 5.147 0 7.756-.854 7.756-2.54 0-1.7-2.609-2.563-7.756-2.563zm0-16.329c3.539 0 6.416 2.779 6.416 6.195 0 3.416-2.877 6.194-6.415 6.194h-.039c-3.53-.01-6.39-2.79-6.379-6.197C3.796 2.778 6.675 0 10.214 0zm0 1.666c-2.586 0-4.691 2.031-4.691 4.529-.009 2.49 2.08 4.52 4.657 4.529l.035.833v-.833c2.586 0 4.69-2.032 4.69-4.529 0-2.498-2.104-4.529-4.69-4.529z"
            fill={color ?? isActive ? text.default : icon.secondary}
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
