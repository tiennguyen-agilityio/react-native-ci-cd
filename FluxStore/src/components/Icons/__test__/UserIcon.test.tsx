import {render, fireEvent, screen} from '@testing-library/react-native';

import {UserIcon} from '../UserIcon';

describe('UserIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<UserIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive', () => {
    const {toJSON} = render(<UserIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<UserIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('user-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<UserIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('user-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
