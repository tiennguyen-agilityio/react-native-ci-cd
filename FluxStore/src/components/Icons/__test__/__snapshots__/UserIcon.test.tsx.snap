// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="user-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={24}
    bbWidth={20}
    fill="none"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 20,
        },
      ]
    }
    vbHeight={24}
    vbWidth={20}
    width={20}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M19.783 18.892c0 3.846-5.462 4.29-9.57 4.29H9.92c-2.617-.006-9.277-.166-9.277-4.313 0-3.767 5.242-4.27 9.318-4.29h.547c2.616.006 9.275.166 9.275 4.313zm-9.57-2.563c-5.148 0-7.758.854-7.758 2.54 0 1.701 2.61 2.563 7.759 2.563 5.147 0 7.756-.854 7.756-2.54 0-1.7-2.609-2.563-7.756-2.563zm0-16.329c3.539 0 6.416 2.779 6.416 6.195 0 3.416-2.877 6.194-6.415 6.194h-.039c-3.53-.01-6.39-2.79-6.379-6.197C3.796 2.778 6.675 0 10.214 0zm0 1.666c-2.586 0-4.691 2.031-4.691 4.529-.009 2.49 2.08 4.52 4.657 4.529l.035.833v-.833c2.586 0 4.69-2.032 4.69-4.529 0-2.498-2.104-4.529-4.69-4.529z"
        fill={
          {
            "payload": 4293454060,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`UserIcon renders correctly (matches snapshot) with isActive 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="user-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={24}
    bbWidth={20}
    fill="none"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 20,
        },
      ]
    }
    vbHeight={24}
    vbWidth={20}
    width={20}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M19.783 18.892c0 3.846-5.462 4.29-9.57 4.29H9.92c-2.617-.006-9.277-.166-9.277-4.313 0-3.767 5.242-4.27 9.318-4.29h.547c2.616.006 9.275.166 9.275 4.313zm-9.57-2.563c-5.148 0-7.758.854-7.758 2.54 0 1.701 2.61 2.563 7.759 2.563 5.147 0 7.756-.854 7.756-2.54 0-1.7-2.609-2.563-7.756-2.563zm0-16.329c3.539 0 6.416 2.779 6.416 6.195 0 3.416-2.877 6.194-6.415 6.194h-.039c-3.53-.01-6.39-2.79-6.379-6.197C3.796 2.778 6.675 0 10.214 0zm0 1.666c-2.586 0-4.691 2.031-4.691 4.529-.009 2.49 2.08 4.52 4.657 4.529l.035.833v-.833c2.586 0 4.69-2.032 4.69-4.529 0-2.498-2.104-4.529-4.69-4.529z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
